### PR TITLE
Allow for using block_api.get_block API

### PR DESCRIPTION
**Pull request by eonwarped**
Prepares for eventual deprecation of condenser_api.get_block.

- Handle new output format (inner block parameter)
- Handles HIVE/HBD nai assets used in transfer op and commentOptions op
- Reformatting beneficiaries in commentOptions to match old API

Moves a few configurable options to config:

- streamerOptions.antiForkBufferMaxSize - int, default value is 2. The number of blocks to check ahead to confirm same fork before sending block for processing 
- streamerOptions.maxQps - int, default value is 1. Max queries to send at a time per node in config 
- streamerOptions.lookaheadBufferSize - int, default value is 5. Number of blocks to fetch at a time using all nodes in config. Note that if any node in the config is unresponsive it can cause the block fetching to stall.
- streamerOptions.hashVerificationMode - string or false, default value is false. Whether to tail a different hive engine API and check for hash match before committing a block to the DB.
- streamerOptions.useBlockApi - boolean, default value is false. Whether to use block_api get_block as opposed to condenser_api.get_block which is deprecated.

Also
- Minor fixes to debuggability (hash verification mode)

### CODE REVIEW

File: plugins/Blockchain.js [hunk 1]
```diff
@@ -74,40 +102,41 @@ async function producePendingTransactions(
...
-      url: 'https://api.hive-engine.com/rpc/blockchain',
-      method: 'POST',
-      headers: {
-        'content-type': 'application/json',
-      },
-      data: {
-        jsonrpc: '2.0', id: 10, method: 'getBlockInfo', params: { blockNumber: newBlock.blockNumber },
-      },
-    })).data.result;
+    const mainBlock = !enableHashVerification ? null : (await getBlock(newBlock.blockNumber));
```
[the remark this answers is not in the document]
Comment (eonwarped) [reply plugins/Blockchain.js]: Aw fine, I was thinking it should more be a secret config. But I suppose as long as everyone is responsible it should be fine.

File: plugins/Blockchain.js [hunk 2]
```diff
@@ -50,6 +50,34 @@ function getRefBlockNumber(block) {
   return block.refHiveBlockNumber;
 }
 
+var gbid = 1;
```
[the remark this answers is not in the document]
Comment (eonwarped) [reply plugins/Blockchain.js:53]: Done

File: plugins/Blockchain.js [hunk 3]
```diff
@@ -50,6 +50,34 @@ function getRefBlockNumber(block) {
...
+async function getBlock(blockNumber, tries = 1) {
+  gbid += 1;
+  try {
+    const block = (await axios({
+      url: 'https://api.hive-engine.com/rpc/blockchain',
```
[the remark this answers is not in the document]
Comment (eonwarped) [reply plugins/Blockchain.js]: Done

**Review by bt-cryptomancer — CHANGES_REQUESTED** [review]
It looks like you've changed the enableHashVerification setting to hashVerificationNode, but forgot to make the corresponding update to config.example.json.

Also, can you please document on the PR what each of the new streamerConfig settings is for, and confirm that the default values you've selected are good values for production use (so node operators can simply copy the example when upgrading)?

**Comment by eonwarped** [comment]
> It looks like you've changed the enableHashVerification setting to hashVerificationNode, but forgot to make the corresponding update to config.example.json.
> 
> Also, can you please document on the PR what each of the new streamerConfig settings is for, and confirm that the default values you've selected are good values for production use (so node operators can simply copy the example when upgrading)?

Done with config. Updating PR.

**Review by bt-cryptomancer — APPROVED** [review]
Thanks, looks good now.

**Pull request merged**